store/sqlstore/thread_store.go

```diff
@@ -217,11 +213,12 @@ func (s *SqlThreadStore) GetThreadsForUser(userId, teamId string, opts model.Get
 		return nil, errors.Wrapf(err, "failed to build subquery to count unread replies when getting threads for user id=%s", userId)
 	}
 
-	query := s.getQueryBuilder().
-		Select(`Threads.*,
-				` + postSliceCoalesceQuery() + `,
-				ThreadMemberships.LastViewed as LastViewedAt,
-				ThreadMemberships.UnreadMentions as UnreadMentions`).
+	query := s.threadsSelectQuery.
+		Column(postSliceCoalesceQuery()).
```

Comment:
Due to this, what is happening is that the `DeleteAt` column is now appearing from both Threads and Posts column. It seems like I was wrong, because somehow the DB is not throwing any error for an ambiguous column reference.

```
EXPLAIN ANALYZE SELECT Threads.PostId, Threads.ChannelId, Threads.ReplyCount, Threads.LastReplyAt, Threads.Participants, COALESCE(Threads.DeleteAt, 0) AS DeleteAt, COALESCE(Posts.Id,'') AS Id,COALESCE(Posts.CreateAt,0) AS CreateAt,COALESCE(Posts.UpdateAt,0) AS UpdateAt,COALESCE(Posts.EditAt,0) AS EditAt,COALESCE(Posts.DeleteAt,0) AS DeleteAt,COALESCE(Posts.IsPinned,false) AS IsPinned,COALESCE(Posts.UserId,'') AS UserId,COALESCE(Posts.ChannelId,'') AS ChannelId,COALESCE(Posts.RootId,'') AS RootId,COALESCE(Posts.OriginalId,'') AS OriginalId,COALESCE(Posts.Message,'') AS Message,COALESCE(Posts.Type,'') AS Type,COALESCE(Posts.Props,'{}') AS Props,COALESCE(Posts.Hashtags,'') AS Hashtags,COALESCE(Posts.Filenames,'[]') AS Filenames,COALESCE(Posts.FileIds,'[]') AS FileIds,COALESCE(Posts.HasReactions,false) AS HasReactions,COALESCE(Posts.RemoteId,'') AS RemoteId, ThreadMemberships.LastViewed as LastViewedAt, ThreadMemberships.UnreadMentions as UnreadMentions, (SELECT COUNT(Posts.Id) FROM Posts WHERE Posts.RootId = ThreadMemberships.PostId AND Posts.CreateAt > ThreadMemberships.LastViewed AND Posts.DeleteAt = $1) AS UnreadReplies FROM Threads JOIN Posts ON Posts.Id = Threads.PostId JOIN ThreadMemberships ON ThreadMemberships.PostId = Threads.PostId JOIN Channels ON Threads.ChannelId = Channels.Id WHERE ThreadMemberships.UserId = $2 AND ThreadMemberships.Following = $3 AND (Channels.TeamId = $4 OR Channels.TeamId = $5) AND (Threads.DeleteAt IS NULL OR Threads.DeleteAt = $6) ORDER BY Threads.LastReplyAt DESC LIMIT 25
```

This is the query. Note the multiple "AS DeleteAt".

Reply:
Nice catch! I'm first adding a test for this, and then will figure out a solution.

Reply:
So at the database level, there is no error: it's perfectly fine to alias your result columns with the same name, so long as you don't rely on those aliases to create ambiguity in the execution of the query:
```
> select 1 as num_employees, 1000 as num_employees;
 num_employees | num_employees
---------------+---------------
             1 |          1000
(1 row)
```

In this case, `sqlx` saw two `DeleteAt` columns and two `DeleteAt` struct properties, and would apparently write the last one it found into the last field it found, leaving the other `0`. I extended the tests to catch this and explicitly verify that the two fields are supposed to have the same value, and then aliased these queries specially to avoid the conflict.